src/renderer/features/network/ManageCustomRpcNode/model/add-custom-rpc-model.ts

```diff
@@ -0,0 +1,178 @@
+import { createEffect, createEvent, createStore, sample } from 'effector';
+import { createForm } from 'effector-forms';
+
+import { networkService } from '@shared/api/network';
+import { ExtendedChain } from '@entities/network';
+import { CheckRpcNodeFxParams, NodeExistParam, RpcCheckResult, SaveRpcNodeFxParams } from '../lib/types';
+import { manageNetworkModel } from '@pages/Settings/Networks/model/manage-network-model';
+import { fieldRules, RpcValidationMapping } from '../lib/constants';
+
+const $addCustomRpcForm = createForm({
+  fields: {
+    name: {
+      init: '',
+      rules: fieldRules.name,
+    },
+    url: {
+      init: '',
+      rules: fieldRules.url,
+    },
+  },
+  validateOn: ['change'],
+});
+
+const formInitiated = createEvent();
+const networkChanged = createEvent<ExtendedChain>();
+const rpcConnectivityChecked = createEvent<RpcCheckResult>();
+const nodeExistChecked = createEvent<boolean>();
+const processStarted = createEvent<boolean>();
+
+const $selectedNetwork = createStore<ExtendedChain | null>(null);
+const $rpcConnectivityResult = createStore<RpcCheckResult>(RpcCheckResult.INIT);
+const $isNodeExist = createStore<boolean>(false);
+const $isProcessStarted = createStore<boolean>(false);
+const $isLoading = createStore<boolean>(false);
+
+const checkRpcNodeFx = createEffect(async ({ network, url }: CheckRpcNodeFxParams): Promise<RpcCheckResult> => {
+  const validationResult = await networkService.validateRpcNode(network.chainId, url);
+  const result = RpcValidationMapping[validationResult];
+
+  return result;
+});
```

Comment:
Better pass `chainId` instead of `network` if that's the only thing we need
Also renamed params/result types - a bit simpler naming
```suggestion
const validateRpcNodeFx = createEffect(async ({ chainId, url }: ValidateRpcParams): Promise<ValidateRpcResult> => {
  const validationResult = await networkService.validateRpcNode(chainId, url);
  
  return RpcValidationMapping[validationResult];
});
```

Reply:
I made the change to pass the result directly, but making the change as you suggest with the `chainId` completely messes up with the sample I have in place, that combines different effects, such as the one with checking node existence, where more than the `chainId` is used.

```ts
sample({
  clock: $addCustomRpcForm.submit,
  source: { network: $selectedNetwork, url: $addCustomRpcForm.fields.url.$value },
  filter: (params: { network: ExtendedChain | null; url: string }): params is VerifyRpcConnectivityFxParams =>
    !!params.network,
  target: [verifyRpcConnectivityFx, isNodeExistFx],
});
```

I know you're saying we shouldn't be scared about duplicating, but given the filtering here, I think this is not worth it. Let me know.